### PR TITLE
Socint 266 use available fulfilment from survey

### PR DESCRIPTION
**Pull request by Paul-Joel**
# Motivation and Context
Switches to use new /surveys endpoint from RHSvc to determine available deliveryMethods for the case(survey), and replaces use of the /fulfilments endpoint for determining appropriate packCode

# What has changed
Adds dynamic function to determine available delivery methods for a fulfilment, determined by the underlying survey data
Adds fallback if no allowableFulfilments provided
Replaces use of /fulfilments endpoint with /survey endpoint to determine 'packCode' for call to RM to request a fulfilment

# How to test?
Start at /en/request/access-code/enter-address/
Proceed through as normal.
Check 'Select delivery method' options are driven by survey
Check appropriate packCode selection is made

# Links
https://collaborate2.ons.gov.uk/jira/browse/SOCINT-266
https://collaborate2.ons.gov.uk/jira/browse/SOCINT-269

### CODE REVIEW

File: app/request_handlers.py [hunk 1]
```diff
@@ -638,14 +707,10 @@ async def post(self, request):
...
-                            fulfilment_code_array.append(fulfilment['fulfilmentCode'])
-                else:
-                    fulfilment_code_array.append(available_fulfilments[0]['fulfilmentCode'])
+                available_fulfilments = \
+                    await RHSvc.survey_fulfilments_by_type(request, 'SMS', survey_id, fulfilment_language)
+
+                fulfilment_code_array.append(available_fulfilments)
 
                 try:
                     await RHSvc.request_fulfilment_sms(request, case_id, mobile_number, fulfilment_code_array)
```
Comment (rpcatling) [inline app/request_handlers.py:716]: can't we do away with `fulfilment_code_array` and just use `available_fulfilments` here ?

File: app/request_handlers.py [hunk 2]
```diff
@@ -843,14 +909,10 @@ async def post(self, request):
...
-                            fulfilment_code_array.append(fulfilment['fulfilmentCode'])
-                else:
-                    fulfilment_code_array.append(available_fulfilments[0]['fulfilmentCode'])
+                available_fulfilments = \
+                    await RHSvc.survey_fulfilments_by_type(request, 'EMAIL', survey_id, fulfilment_language)
+
+                fulfilment_code_array.append(available_fulfilments)
 
                 try:
                     await RHSvc.request_fulfilment_email(request, case_id, email, fulfilment_code_array)
```
Comment (rpcatling) [inline app/request_handlers.py:918]: can't we do away with `fulfilment_code_array` and just use `available_fulfilments` here ?

File: app/request_handlers.py [hunk 3]
```diff
@@ -1051,19 +1114,10 @@ async def post(self, request):
...
-                if len(available_fulfilments) > 1:
-                    for fulfilment in available_fulfilments:
-                        if fulfilment['language'] == fulfilment_language:
-                            fulfilment_code_array.append(fulfilment['fulfilmentCode'])
-                else:
-                    fulfilment_code_array.append(available_fulfilments[0]['fulfilmentCode'])
+                available_fulfilments = \
+                    await RHSvc.survey_fulfilments_by_type(request, 'POST', survey_id, fulfilment_language)
+
+                fulfilment_code_array.append(available_fulfilments)
```
Comment (rpcatling) [inline app/request_handlers.py:1120]: can't we do away with `fulfilment_code_array` and just use `available_fulfilments` here ?

**Review by rpcatling — APPROVED** [review]
just suggestion for slightly simpler code .

**Review by philwhiles — APPROVED** [review]

**Pull request merged**